File: 121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int p=0;
        if (prices.size()==0)
            return p;
        int a=prices[0];
        for (int i=0;i<prices.size();i++)
        {
            if (prices[i]<a)
                a=prices[i];
            else if (prices[i]-a>p)
                p=prices[i]-a;
        }
        return p;
    }
};

File: 121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices)==0:
            return 0
        b=prices[0]
        a=0
        for i in prices:
            if i<b:
                b=i
            if i-b>a:
                a=i-b
        return a
